app/js/states/inventory/inventory.controller.js

```diff
@@ -298,40 +300,40 @@ function InventoryCtrl(
     function order() {
         $scope.systems = $filter('orderBy')($scope.systems,
             ['_type',
-            ($scope.reverse ?
-                '-' + $scope.predicate :
-                $scope.predicate)]);
+            ($scope.sorter.reverse ?
+                '-' + $scope.sorter.predicate :
+                $scope.sorter.predicate)]);
     }
 
     $scope.sort = function (column) {
         $scope.loading = true;
 
-        // just changing the sort direction
-        if (column === InventoryService.getSortField()) {
-            InventoryService.toggleSortDirection();
-            $scope.reverse = !$scope.reverse;
-        }
-        else {
-            InventoryService.setSortField(column);
-            InventoryService.setSortDirection('ASC');
-            $scope.reverse = false;
-            $scope.predicate = column;
-
-            // special case where we are sorting by timestamp but visually
-            // showing timeago
-            if (column === 'last_check_in') {
-                InventoryService.setSortDirection('DESC');
-            }
+        // special case where we are sorting by timestamp but visually
+        // showing timeago
+        if (column === 'last_check_in' &&
+                !$scope.sorter.changeReverse) {
+            $scope.sorter.changeReverse = true;
+        } else if ($scope.sorter.changeReverse) {
+            $scope.sorter.changeReverse = false;
         }
 
         // if we have the full inventory list then use local sorting
         if ($scope.systems.length === InventoryService.getTotal()) {
-            order();
+            if ($scope.sorter.disableCallback) {
+                $scope.sorter.disableCallback = false;
+            }
+
+            $scope.sorter.sort(column);
             $scope.loading = false;
```

Comment:
let's just get rid of the local sorting for now. It's not working anyway. If I get all the systems and sort, I still see an api call going out. This should remove the need for a disableCallback flag.